From: Dermot Bradley <dermot_bradley@yahoo.com>
Date: Fri, 28 Aug 2020 19:26 +0100
Subject: [PATCH] step-certificates: Build Yubikey KMS binary

Enable building of Yubikey KMS binary, to do this also enable CGO.

---

diff -aur a/Makefile b/Makefile
--- a/Makefile
+++ b/Makefile
@@ -86,7 +86,7 @@
 
 DATE    := $(shell date -u '+%Y-%m-%d %H:%M UTC')
 LDFLAGS := -ldflags='-w -X "main.Version=$(VERSION)" -X "main.BuildTime=$(DATE)"'
-_GOFLAGS := CGO_ENABLED=0
+_GOFLAGS := CGO_ENABLED=1
 
 download:
 	$Q go mod download
@@ -167,15 +167,19 @@
 
 INSTALL_PREFIX?=/usr/
 
-install: $(PREFIX)bin/$(BINNAME) $(PREFIX)bin/$(CLOUDKMS_BINNAME) $(PREFIX)bin/$(AWSKMS_BINNAME)
+install: $(PREFIX)bin/$(BINNAME) $(PREFIX)bin/$(CLOUDKMS_BINNAME) $(PREFIX)bin/$(AWSKMS_BINNAME) $(PREFIX)bin/$(YUBIKEY_BINNAME) $(PREFIX)bin/$(PKCS11_BINNAME)
 	$Q install -D $(PREFIX)bin/$(BINNAME) $(DESTDIR)$(INSTALL_PREFIX)bin/$(BINNAME)
 	$Q install -D $(PREFIX)bin/$(CLOUDKMS_BINNAME) $(DESTDIR)$(INSTALL_PREFIX)bin/$(CLOUDKMS_BINNAME)
 	$Q install -D $(PREFIX)bin/$(AWSKMS_BINNAME) $(DESTDIR)$(INSTALL_PREFIX)bin/$(AWSKMS_BINNAME)
+	$Q install -D $(PREFIX)bin/$(YUBIKEY_BINNAME) $(DESTDIR)$(INSTALL_PREFIX)bin/$(YUBIKEY_BINNAME)
+	$Q install -D $(PREFIX)bin/$(PKCS11_BINNAME) $(DESTDIR)$(INSTALL_PREFIX)bin/$(PKCS11_BINNAME)
 
 uninstall:
 	$Q rm -f $(DESTDIR)$(INSTALL_PREFIX)/bin/$(BINNAME)
 	$Q rm -f $(DESTDIR)$(INSTALL_PREFIX)/bin/$(CLOUDKMS_BINNAME)
 	$Q rm -f $(DESTDIR)$(INSTALL_PREFIX)/bin/$(AWSKMS_BINNAME)
+	$Q rm -f $(DESTDIR)$(INSTALL_PREFIX)/bin/$(YUBIKEY_BINNAME)
+	$Q rm -f $(DESTDIR)$(INSTALL_PREFIX)/bin/$(PKCS11_BINNAME)
 
 .PHONY: install uninstall
 
